rmw_connextdds_common/src/common/rmw_impl.cpp

```diff
@@ -1730,7 +1889,7 @@ rmw_connextdds_create_subscriber(
       "failed to allocate RMW_Connext_Subscriber")
     return nullptr;
   }
-
+  rmw_sub_impl->set_node(node);
```

Comment:
After some further consideration, I would remove `RMW_Connext_Subscriber::set_node()` and instead add a `node` argument to both `RMW_Connext_Subscriber::create()` and `RMW_Connext_Subscriber::RMW_Connext_Subscriber()`.

The only reason, I'm assuming, why it wasn't added as a parameter in the first place is because those functions are also used to create "internal" subscribers (e.g. in `rmw_graph.cpp`), which do not belong to any node.

I think it's more consistent to make the "pointer to the parent node" a creation argument, to make it clear that it's something that it's not going to change over the lifetime of the subscriber, and pass `nullptr` for it when creating an `internal` endpoint, rather than adding a setter for the field, which suggests "mutability", even though the field is not expected to change.

I suggest adding a `const rmw_node_t * const node` parameter to `RMW_Connext_Subscriber::create()` and `RMW_Connext_Subscriber::RMW_Connext_Subscriber()`, and then set `node` in the initialization list in `RMW_Connext_Subscriber::RMW_Connext_Subscriber()`.

Reply:
Calls to `RMW_Connext_Subscriber::create()` which do not have a `node` available (e.g. in `rmw_graph.cpp`) should pass `nullptr /* node */`.